README.md

```diff
@@ -1,16 +1,29 @@
-# humor-personalization
-Personalized sense of humor prediction
+Project: nlperspectives
 
-## Download data
+Method: USER_ID
 
-To download data, enter personalized_nlp folder and type in:
 
-`dvc pull`
+## Important files in this branch:
 
-## How to run experiments:
+- personalized_nlp/models/user_id.py	: Code for the 'userid' method (add the annotator_id as a special token on the embeddings without modifying the raw_texts or annotations)
 
-`python -m personalized_nlp.experiments.jester`
+- personalized_nlp/models/__init__.py	: Added mapping for the 'userid' method
 
-## How to add new dataset
+- personalized_nlp/settings.py		: Added EMBEDDINGS_SIZES and TRANSFORMER_MODEL_STRINGS for 'roberta' (specifically, roberta-base)
 
-Copy one of existings dataset classes (personalized_nlp/datasets/) and modify paths and settings. Next, copy one of the experiments (personalized_nlp/experiments/) and customize the settings.
\ No newline at end of file
+- personalized_nlp/experiments/emopers_userid_notune.py	: A test-drive experiment without fine-tuning
+
+- personalized_nlp/experiments/emopers_userid_tune.py		: The experiment involving fine-tuning (also here: https://colab.research.google.com/drive/1eZqlMPia3Vxn10kxosI3NgyMJ93uIyAX?usp=sharing)
+
+- personalized_nlp/learning/train_alt.py : A customized version of train.py where a flag is added to choose running against test split or not, also added a method for getting the checkpoint directory
+
+## To Do:
+
+- a preprocessing step of averaging the embedding with attention mask > 0 before feeding into fc1
+
+## Questions:
+When I try to use some code from embeddings.py to average the embedding with attention mask > 0, (i.e. the line "mask = batch_encoding attention_mask > 0" and the loop "for i in range(emb....):....", I get the error saying "there are too many indices for tensor of dimension 1".
+
+I tried changing the line "all_embeddings.append(emb\[0\]\[i, mask\[i\] > 0, :\].mean(axis=0)\[None, :\])" into "all_embeddings.append(emb\[0\]\[mask\[i\]].mean(axis=0)\[None, :\])" and I get another error saying that the mask's shape \[128\] does not match the tensor's shape \[768\].
+
+How to do this correctly?
```

Comment:
Hmm, I have not met that problem so I did not toch that code. Do you still have it? If I understand right, you couldn't get the pretrained embeddings for texts. I think, it would be easier if you show me that during some debugging.

Reply:
Sorry for not updating the readme; I have deleted that piece of code. I wrote that question when Anh told me that I should "check the /utils/embedding.py, in function _get_embeddings() before concatenating all the embeddings, there's a preprocessing step of averaging the embedding with attention mask > 0. All of the other text embeddings we create for other models (using data module) are applied to this processing. Therefore, I think we can add that additional part also, before concat them and feeding to fc1."

From what I understand, I am supposed to do the averaging of the embeddings with attention mask > 0 before feeding them to fc1. But I was not able to do that, as I kept getting errors that I didn't really understand 😓 . So I deleted the piece of code for averaging the embeddings with attention mask > 0, and the model seemed to run fine. In your opinion, should I implement this averaging pre-processing?

Reply:
I have debugged it and the resulting avereged emebddings looks the same in both verions:
![image](https://user-images.githubusercontent.com/33195859/162448135-72beeb7a-4e8c-4329-8566-93fc39e6e0b7.png)

So if your version is the same as for that `avg1` in the screenshot, it is ok. Still, I don't know why it is not working for ```emb[0][i, mask[i] > 0, :].mean(axis=0)[None, :]```. If you want you can check if you have the same error for running experiment for jester.